Core/src/Geometry/TrackingVolume.cpp

```diff
@@ -507,7 +509,7 @@ Acts::TrackingVolume::compatibleBoundaries(
 
     ACTS_VERBOSE("Check intersection with surface "
                  << bSurface->surfaceRepresentation().geometryId());
-    if (detail::checkIntersection(sIntersection.intersection, pLimit, oLimit,
+    if (detail::checkIntersection<decltype(sIntersection.intersection), decltype(logger)>(sIntersection.intersection, pLimit, oLimit,
```

Comment:
@paulgessinger - this (and few lines below) call got a bit uglier. The compiles is not able to deduce that we want to pass by ref w.o these explicit parametrisation. It assumes we want to pass longer by copy and that is disallowed. 
What go you think? Does the loger parameter in the checkIntersection need to be templated parameter?

Reply:
Is it not able to deduct the logger type?

Reply:
Also, from looking at it, no I don't think it needs to be templated, the whole mechanism checking the type of the logger can probably go away. Do you want to make this change in this PR, or should this be a follow up PR?

Reply:
Do you mean that the logger can be a concrete type? Sure I can do it.

Reply:
Actually that change cause avalanche of changes that should be done for clients, and client of clients .... 
I think it is better to do it as a followup.

Reply:
Ok!